src/main/java/io/cryostat/core/agent/Event.java

```diff
@@ -0,0 +1,277 @@
+    }
+
+    private static Element getFirstDirectChildElementByTagName(Element parent, String name) {
+        NodeList children = parent.getChildNodes();
+        for (int i = 0; i < children.getLength(); i++) {
+            Node child = children.item(i);
+            if (child instanceof Element && child.getNodeName().equals(name)) {
+                return (Element) child;
+            }
+        }
+
+        return null;
+    }
+
+    private MethodParameter createMethodParameter(Element element) {
+        return new MethodParameter(this, element);
+    }
+
+    private Field createField(Element element) {
+        return new Field(this, element);
+    }
+
+    private MethodReturnValue createMethodReturnValue(Element element) {
+        return new MethodReturnValue(this, element);
+    }
+
+    public Element buildElement(Document document) {
+        Element element = document.createElement(XML_TAG_EVENT);
+        element.setAttribute(XML_ATTRIBUTE_ID, id);
+
+        Element nameElement = document.createElement(XML_TAG_NAME);
+        nameElement.setTextContent(name);
+        element.appendChild(nameElement);
+
+        if (description != null && !description.isEmpty()) {
+            Element descriptionElement = document.createElement(XML_TAG_DESCRIPTION);
+            descriptionElement.setTextContent(description);
+            element.appendChild(descriptionElement);
+        }
+
+        Element classElement = document.createElement(XML_TAG_CLASS);
+        classElement.setTextContent(clazz);
+        element.appendChild(classElement);
+
+        if (path != null && !path.isEmpty()) {
+            Element pathElement = document.createElement(XML_TAG_PATH);
+            pathElement.setTextContent(path);
+            element.appendChild(pathElement);
+        }
+
+        Element stackTraceElement = document.createElement(XML_TAG_STACK_TRACE);
+        stackTraceElement.setTextContent(String.valueOf(recordStackTrace));
+        element.appendChild(stackTraceElement);
+
+        Element rethrowElement = document.createElement(XML_TAG_RETHROW);
+        rethrowElement.setTextContent(String.valueOf(useRethrow));
+        element.appendChild(rethrowElement);
+
+        if (location != null) {
+            Element locationElement = document.createElement(XML_TAG_LOCATION);
+            locationElement.setTextContent(location.toString());
+            element.appendChild(locationElement);
+        }
+
+        element.appendChild(buildMethodElement(document));
+
+        if (!fields.isEmpty()) {
+            Element fieldsElement = document.createElement(XML_TAG_FIELDS);
+            for (Field field : fields) {
+                fieldsElement.appendChild(field.buildElement(document));
+            }
+            element.appendChild(fieldsElement);
+        }
+
+        return element;
+    }
+
+    private Element buildMethodElement(Document document) {
+        Element methodElement = document.createElement(XML_TAG_METHOD);
+        {
```

Comment:
I would prefer to remove this scope block - it doesn't seem to be doing much lexical separation of scopes, so it's just visual noise to me